### ISSUE TITLE
Istio module migration

### ISSUE THREAD

**Issue by strekm** [kyma-project/istio#168]
**Description**

Based on [POC](https://github.com/kyma-project/istio/issues/148) prepare migration procedure for Istio module. In Istio module introduce alpha2 version of Istio CRD to include kyma release version. Storage version should be alpha2 version. Implement migration script executing following tasks:
- check if istio cr already exists
- add istio module with customResourcePolicy: Ignore to kyma reasource (install operator + CRD)
- apply istio CR if doesn't exist (do not use customResourcePolicy: Ignore)

Script should be implemented as a bash script that later on will be executed as a task on managed kyma environments. Applied Istio CR should not include any k8s resource configuration. If there's existing Istio CR containing configuration for numTrustedProxies the script should not change that. 
Provide documentation of migration process. 
Configure istio-reconciler to stop working if there's alpha2 on a cluster

ACs:
- [x] Istio CRD updated to alpha2 version 
- [x] CRD contains alpha1 version
- [x] alpha2 is storage version
- [x] migration script implemented
- [x] documentation
- [x] tested on managed kyma dev environment
- [x] tested on managed kyma dev environment after exclusion PR and manager changes are merged
  - [x] Scenario SKR with existing Istio CR 
  - [x] Scenario SKR without existing Istio CR
  - [x] Scenario SKR with an existing module
- [x] istio reconciler configuration exists
- [x] Clarify how to continue with CRDs component error after migration

**PRs**
- https://github.com/kyma-project/control-plane/pull/2847
- https://github.com/kyma-project/istio/pull/176

**Reasons**

modularisation

**DoD:**
- [x] provide documentation
~~- [ ] release notes and What's New updates for Kyma customers~~
~~- [ ] provide unit tests~~
~~- [ ] provide integration tests~~
- [x] test on production-like environment
- [x] verify resource limits
~~- [ ] followup issue~~
~~- [ ] PR reviewer will verify code coverage and evaluate if it is acceptable~~

**Attachments**

part of: https://github.com/kyma-project/istio/issues/38

**Comment by triffer** [comment]
There can be a race condition during the module migration from reconciler to Lifecycle-Manager.
I don't think there won't be an issue if this happens during our migration since the reconciler and the module will use the same Istio version and also Istio CR won't be changed during the migration. Since we'll be doing the migration on Stage together with the SRE team, they might already have an opinion about this due to experience from other module migrations.

**Comment by videlov** [comment]
Should we disable Reconciler eventually, perhaps consult with SRE team and could also adapt migration script?

**Comment by triffer** [comment]
While testing this feature on SKR Dev after rolling out the KCP configuration changes, we encountered two issues.

1. The deployment of istio-manager is blocked by the warden admission controller, since the pod validation failed. The root cause is that the validation of `gcr.io/kubebuilder/kube-rbac-proxy` failed.
2. After applying the CRD from our Istio module the reconciliation of istio by the old reconciler should stop. This is working, istio is no longer reconciled. But the CRDs component is in an error state and the following warning logged:
```
{"level":"WARN","time":"2023-06-19T09:30:34.649Z","message":"kubeClient failed to update CustomResourceDefinition 'istios.operator.kyma-project.io' (namespace: )  with strategy 'PATCH': cannot patch \"istios.operator.kyma-project.io\" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io \"istios.operator.kyma-project.io\" is invalid: status.storedVersions[1]: Invalid value: \"v1alpha2\": must appear in spec.versions","correlation-id":"433753ff-1d0b-4ff7-a7f3-8f8780ddd3c1--2030b7b8-e265-4fe7-9a5f-db13c14c6a1d","component-name":"CRDs"}
```
With Istio-module we roll out version v1alpha2 for istios.operator.kyma-project.io. The version v1alpha2 is also our trigger to disable the istio component reconciliations. We have to use v1alpha2 since v1alpha1 is already rolled out and can be used by the customer.
The crd in the kyma repository doesn't include the v1alpha2, so the reconciliation of the istio CRDs should be disabled, too.

**Comment by tobiscr** [comment]
Couldn't we have both CRD versions available at the same time? If yes, we could add the v1 and v2 version to the CRD-component of the old reconciler and we shouldn't face this issue.

**Comment by triffer** [comment]
The issue with CRDs component will be handled in issue:
- https://github.com/kyma-incubator/reconciler/issues/1362
